### PR TITLE
make it possible to call backend without a net

### PR DESCRIPTION
**Pull request by borg323**
Changes all backends to make the WeightsFile passed to them optional, with an error if not there in all except random.
Also included is the hack to fix the c++ version incompatibility with tensorflow.

### CODE REVIEW

File: src/neural/factory.cc [hunk 1]
```diff
@@ -123,7 +123,10 @@ std::unique_ptr<Network> NetworkFactory::LoadNetwork(
   } else {
     CERR << "Loading weights file from: " << net_path;
   }
-  const WeightsFile weights = LoadWeightsFromFile(net_path);
+  std::optional<WeightsFile> weights;
+  if (!net_path.empty()) {
```
Comment (Tilps) [inline src/neural/factory.cc:127]: This means random backend needs an explicit --weights= since the default is auto discover.
But I'm fine with that...

Comment (Tilps) [reply src/neural/factory.cc:127]: Err nevermind, I can't read.

File: src/neural/network_demux.cc [hunk 2]
```diff
@@ -235,8 +235,12 @@ void DemuxingComputation::ComputeBlocking() {
   dataready_cv_.wait(lock, [this]() { return dataready_ == 0; });
 }
 
-std::unique_ptr<Network> MakeDemuxingNetwork(const WeightsFile& weights,
-                                             const OptionsDict& options) {
+std::unique_ptr<Network> MakeDemuxingNetwork(
```
Comment (Tilps) [inline src/neural/network_demux.cc:238]: Technically these ones could defer failure to the child networks, but I'm fine with it not - given the unlikelyhood of people using a roundrobin/demux/mux of random backends more than once...

Comment (borg323) [reply src/neural/network_demux.cc:238]: I wanted to have a first version ready quickly.

Comment (Tilps) [reply src/neural/network_demux.cc:240]: update looks good.

**Comment by Mardak** [comment]
Thanks. Ran into this when testing #1181 😉

**Pull request merged**